### PR TITLE
UI improvements for the edit mode

### PR DESCRIPTION
**Pull request by vinay033**
### :nut_and_bolt: Description: What code changed, and why?
UX identified some design issues in edit mode, this PR covered some of UX improvement changes.

### :chains: Related Resources
https://github.com/chef/automate/issues/4808
### :+1: Definition of Done
I have covered UX improvements which are mentioned below
1. Delete Data Bag - Button text: “Delete” ->  change to delete
2. Create/Remove Chef Server - The button on modal dialogue says “Delete” where on the drop down it says “Remove” we should make these the same. -  changed to delete
3. When you hover policy files, details page- the hover effect has some glitch. it not coverup the beta tag.
4. Client list page - Create client button misaligned
5. Delete popup modal width issue
6. Roles list page is misaligned when there is no data.
7. FIx pagination issue for the clients
### :athletic_shoe: How to Build and Test the Change
```
Go to the `infrastructure tab` from the top navigation bar then you can see the `Chef Server`
If you have data then you can see the list of servers --> then you can check all the mentioned points.
```
### :white_check_mark: Checklist

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [ ] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/automate-ui/src/app/modules/infra-proxy/create-chef-server-modal/create-chef-server-modal.component.html [hunk 1]
```diff
@@ -1,90 +1,90 @@
...
+      </div>
+      <div *ngIf="!modifyID" class="id-margin">
+        <div id="id-fields">
+          <span class="key-label">ID:&nbsp;</span>
+          <span data-cy="id-label" class="object-id">{{ this.createForm?.value.id }}</span>
         </div>
-      </form>
-    </div>
+        <chef-toolbar>
+          <chef-button tertiary (click)="modifyID = true" id="edit-button-object-modal" data-cy="edit-button">Edit ID</chef-button>
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/create-chef-server-modal/create-chef-server-modal.component.html:45]: This is missing a focus ring around it when it is keyboard focused

File: components/automate-ui/src/app/modules/infra-proxy/create-client-modal/create-client-modal.component.scss [hunk 2]
```diff
@@ -81,7 +85,7 @@ chef-modal {
           font-size: 12px;
           line-height: 20px;
           letter-spacing: 0.25px;
-          width: 80%;
+          width: calc(100% - 45px);
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/create-client-modal/create-client-modal.component.scss:88]: nice!

File: components/automate-ui/src/app/modules/infra-proxy/delete-infra-object-modal/delete-infra-object-modal.component.scss [hunk 3]
```diff
@@ -4,13 +4,16 @@ chef-modal {
...
+    position: absolute;
+    right: 32px;
+    bottom: 32px;
 
     .cancel-button {
       color: $chef-critical;
     }
   }
 
+  // margin-top: -32px;
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/delete-infra-object-modal/delete-infra-object-modal.component.scss:16]: left behind comment

**Review by SEAjamieD — APPROVED** [review]
Great job on these fixes - all are working well for me 🎉 

I noticed that on many of the delete modals, I'm seeing the content break out of the container when the browser is collapsed.  Would be ideal to be fixed here but could also be addressed elsewhere.

Delete Organization modal break out

**Review by kalroy — APPROVED** [review]
Looks good to me.

**Comment by vinay033** [comment]
> Great job on these fixes - all are working well for me 🎉
> 
> I noticed that on many of the delete modals, I'm seeing the content break out of the container when the browser is collapsed. Would be ideal to be fixed here but could also be addressed elsewhere.
> 
> Delete Organization modal break out

I have added some more changes for the modals, when you get the chance please check it once.

**Comment by SEAjamieD** [comment]
> > Great job on these fixes - all are working well for me 🎉
> > I noticed that on many of the delete modals, I'm seeing the content break out of the container when the browser is collapsed. Would be ideal to be fixed here but could also be addressed elsewhere.
> > Delete Organization modal break out
> 
> I have added some more changes for the modals, when you get the chance please check it once.

Thanks for the update Vinay - Ideally, the modal should still be flexible like before and the content inside is adjusted to fit as it shrinks.  I understand that the modal content designs are new and in flux right now, so I think this solves the issue for the time being 👍

**Pull request merged**